.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,144 @@
+name: build-and-test
+on:
+  push:
+    branches: [master]
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+  pull_request:
+
+jobs:
+  windows-test:
+    runs-on: windows-latest
+    env:
+      TEST_RESULTS: testbed/tests/results/junit/results.xml
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
+      - name: Setup env
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+        shell: bash
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: \Users\runneradmin\go\pkg\mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
+      - name: Run Unit tests
+        run: go test ./...
+      - name: GitHub issue generator
+        if: ${{ failure() && github.ref == 'ref/head/master' }}
+        run: |
+          go run cmd/issuegenerator/main.go $TEST_RESULTS
+  check-links:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Node
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: '9'
+      - name: Check links in markdown docs
+        run: |
+          pushd $HOME
+          npm install --save-dev markdown-link-check@3.8.1
+          popd
+      - run: ./.github/workflows/check-links/check-links.sh
```

Comment:
I would move this in a separate action. This should not block a release or any testing/releasing steps, only merging :)

Reply:
oh ok i see. So `check_links` should only be run on an open pull request?

Reply:
Can be run on master as well, but should not block other actions if possible